From haiyangz@microsoft.com  Thu Apr 22 16:42:47 2010
From: Haiyang Zhang <haiyangz@microsoft.com>
Date: Mon, 19 Apr 2010 15:32:11 +0000
Subject: Staging: hv: Fix a bug affecting IPv6
To: Greg KH <gregkh@suse.de>
Cc: Hank Janssen <hjanssen@microsoft.com>, Toshikazu Sakai <Toshikazu.Sakai@microsoft.com>
Message-ID: <1FB5E1D5CA062146B38059374562DF7266281C0D@TK5EX14MBXC121.redmond.corp.microsoft.com>


From: Haiyang Zhang <haiyangz@microsoft.com>

Fix a bug affecting IPv6
Added the multicast flag for proper IPv6 function.

Reported-by: Toshikazu Sakai <toshikas@microsoft.com>
Signed-off-by: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Haiyang Zhang <haiyangz@microsoft.com>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/hv/RndisFilter.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/staging/hv/RndisFilter.c
+++ b/drivers/staging/hv/RndisFilter.c
@@ -751,6 +751,7 @@ static int RndisFilterOpenDevice(struct
 
 	ret = RndisFilterSetPacketFilter(Device,
 					 NDIS_PACKET_TYPE_BROADCAST |
+					 NDIS_PACKET_TYPE_ALL_MULTICAST |
 					 NDIS_PACKET_TYPE_DIRECTED);
 	if (ret == 0)
 		Device->State = RNDIS_DEV_DATAINITIALIZED;
